### PR TITLE
Allow to specify element into which to inject Style.

### PR DESCRIPTION
**Pull request by unity**
We use this to allow injecting a Style tag into a different place than HEAD.

We use React+Webpack a lot to bundle reusable applications embedded in a 3rd party page.
Here are two examples of how we benefit from this PR:
- To to inject the style into a virtual Iframe (with https://github.com/ryanseddon/react-frame-component or some other tech we're still working on)
- To inject the style in a specific place in the DOM, so we can monitor it with Mutation Observers

### CODE REVIEW

**Comment by vvo** [comment]
HI @unity How do you pass the corresponding element in the webpack config? Thanks

**Comment by unity** [comment]
You pass it at runtime : 

```js
import style from './style.css'

style.use({
  element:document.getElementById('myId');
});
```

**Comment by mmoutenot** [comment]
is there any chance this will be merged? Would love to use this for iframes

**Comment by ronkorving** [comment]
Currently this PR has conflicts. Rebasing would be a good start. Would like to see this merged too.

File: addStyles.js [hunk 1]
```diff
@@ -95,11 +95,11 @@ function listToStyles(list) {
 	return styles;
 }
 
-function createStyleElement() {
+function createStyleElement(element) {
 	var styleElement = document.createElement("style");
```
Comment (michael-ciniawsky) [inline addStyles.js]: `styleElement` => `style`

Comment (michael-ciniawsky) [inline addStyles.js]: `parentElement` => `element`

Comment (michael-ciniawsky) [inline addStyles.js]: and all following references to them please :)

File: addStyles.js [hunk 2]
```diff
@@ -13,9 +13,9 @@ var stylesInDom = {},
 	isOldIE = memoize(function() {
 		return /msie [6-9]\b/.test(window.navigator.userAgent.toLowerCase());
 	}),
-	getHeadElement = memoize(function () {
```
Comment (ekulabuhov) [inline addStyles.js]: I don't see a reason to remove memoization.

File: addStyles.js [hunk 3]
```diff
@@ -13,9 +13,9 @@ var stylesInDom = {},
...
-		return document.head || document.getElementsByTagName("head")[0];
-	}),
+	getElement = function (element) {
```
Comment (ekulabuhov) [inline addStyles.js]: I don't see a reason to remove memoization.

**Comment by michael-ciniawsky** [comment]
Superseded by #135 Thx 😛

**Comment by Aqours** [comment]
```js
import style from './style.css'
style.use(parent.document.getElementById('myId'));
```
IMO, this way is more useful than set config in style-loader. Consider to add this feature

**Pull request closed without merging**